File: Serene3/Serene3.Web/Modules/com/orderdetials/orderdetialsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Serene3.com.Columns;

[ColumnsScript("com.orderdetials")]
[BasedOnRow(typeof(orderdetialsRow), CheckNames = true)]
public class orderdetialsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ProductidName { get; set; }
    public decimal Totalprice { get; set; }
    public int Quantity { get; set; }
    public string OrderidUserid { get; set; }
    public decimal Price { get; set; }
    public DateOnly Entitydata { get; set; }
}

File: Serene3/Serene3.Web/Modules/ServerTypes/com/orderdetialsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { orderdetialsRow } from "./orderdetialsRow";

export interface orderdetialsColumns {
    Id: Column<orderdetialsRow>;
    ProductidName: Column<orderdetialsRow>;
    Totalprice: Column<orderdetialsRow>;
    Quantity: Column<orderdetialsRow>;
    OrderidUserid: Column<orderdetialsRow>;
    Price: Column<orderdetialsRow>;
    Entitydata: Column<orderdetialsRow>;
}

export class orderdetialsColumns extends ColumnsBase<orderdetialsRow> {
    static readonly columnsKey = 'com.orderdetials';
    static readonly Fields = fieldsProxy<orderdetialsColumns>();
}